### PR TITLE
Add pre-commit hook configuration

### PR DESCRIPTION
**Pull request by burakince**
This pull-request is implementation of issue #279

Currently, main branch (Commit hash: 0c139521a680b22f1bcbb08bab5250d4197a1265) and tag [v0.10.8](https://github.com/datacontract/datacontract-cli/releases/tag/v0.10.8) are getting the following error.

```bash
Traceback (most recent call last):
  File "/var/folders/pb/3h4kqhbs01g8qqhj9vpp8kvc0000gn/T/tmplr11kfy7/repo36bh8g_4/py_env-python3.12/bin/datacontract", line 5, in <module>
    from datacontract.cli import app
  File "/private/var/folders/pb/3h4kqhbs01g8qqhj9vpp8kvc0000gn/T/tmplr11kfy7/repo36bh8g_4/py_env-python3.12/lib/python3.12/site-packages/datacontract/cli.py", line 16, in <module>
    from datacontract import web
  File "/private/var/folders/pb/3h4kqhbs01g8qqhj9vpp8kvc0000gn/T/tmplr11kfy7/repo36bh8g_4/py_env-python3.12/lib/python3.12/site-packages/datacontract/web.py", line 7, in <module>
    from datacontract.data_contract import DataContract, ExportFormat
  File "/private/var/folders/pb/3h4kqhbs01g8qqhj9vpp8kvc0000gn/T/tmplr11kfy7/repo36bh8g_4/py_env-python3.12/lib/python3.12/site-packages/datacontract/data_contract.py", line 7, in <module>
    from pyspark.sql import SparkSession
ModuleNotFoundError: No module named 'pyspark'
```

On the other hand, tag [v0.10.7](https://github.com/datacontract/datacontract-cli/releases/tag/v0.10.7) works well. You can find an example from [here](https://github.com/burakince/datacontract-cli/blob/v0.10.7-rc1/.pre-commit-hooks.yaml).

Simply, you can use the following configuration in your `.pre-commit-config.yaml` and test the idea.

```yaml
repos:
  - repo: https://github.com/burakince/datacontract-cli
    rev: "v0.10.7-rc1"
    hooks:
      - id: datacontract-linting
```

Additionally, you can test your local pre-commit hook changes in an example test repository. Please follow these steps:

* Please first, [install](https://pre-commit.com/#installation) pre-commit to your system.
* Clone the original cli repository with `git clone git@github.com:datacontract/datacontract-cli.git` command.
* And create another folder and git repository inside of the test repo. `mkdir test-repo && cd test-repo && git init`.
* Add an example data contract YAML file inside of test repository.
* Run the following command in test repository and it must use settings from other local `datacontract-cli` folder.

```bash
pre-commit try-repo ../datacontract-cli datacontract-linting --verbose --all-files
```

- [x] Tests pass
- [x] README.md updated
- [x] CHANGELOG.md entry added

### CODE REVIEW

**Comment by jochenchrist** [comment]
> Currently, main branch (Commit hash: https://github.com/datacontract/datacontract-cli/commit/0c139521a680b22f1bcbb08bab5250d4197a1265) and tag [v0.10.8](https://github.com/datacontract/datacontract-cli/releases/tag/v0.10.8) are getting the following error.


With v0.10.8, the library uses [extras](https://cli.datacontract.com/#optional-dependencies). 
You can fix it with `pip install datacontract-cli[all]`

**Comment by burakince** [comment]
Hi @jochenchrist ,

I'd like to explain how pre-commit works. Generally, you don't need to install anything except pre-commit itself for pre-commit hooks. It automatically installs all the required packages as needed. In our case, it attempts to install the `datacontract-cli` Python package, but it seems that additional packages (e.g., `pyspark`) are causing installation issues. These extra packages shouldn't be required for the default installation, so there might be a misconfiguration somewhere. Notably, version v0.10.7 works perfectly fine.

In short, users never need to run the `pip install datacontract-cli` command for the pre-commit system. Pre-commit installs it in the background.

Best regards,
Burak

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+- id: datacontract-linting
```
Comment (jochenchrist) [inline .pre-commit-hooks.yaml:1]: It would be a little more straight-forward to use `datacontract-lint` and `datacontract-test`, to use the same command names as in the cli. 
Or is there a convention to use the gerund?

Comment (burakince) [reply .pre-commit-hooks.yaml:1]: No, there is not specific rule behind it. We can give any id whatever we want. I agree, let's change to `datacontract-lint` and `datacontract-ltest` and people can use them as below:

```yaml
repos:
  - repo: https://github.com/datacontract/datacontract-cli
    rev: "v0.10.0"
    hooks:
      - id: datacontract-lint
      - id: datacontract-test
         args: ["--server", "http://example.com"]
```

**Comment by burakince** [comment]
These definitions only for Python usage. What do you think @jochenchrist , shall we also define id for the docker image usage?

**Comment by burakince** [comment]
Hi @jochenchrist ,

I have added all extras as additional dependencies and tested them locally. Everything is working fine now. We should consider finding a better solution in the future.

Best regards,
Burak

**Comment by jochenchrist** [comment]
OK, everything's fine now, thanks for your contribution!
